client/components/theme/index.jsx

```diff
@@ -62,6 +61,10 @@ var Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
+	shouldComponentUpdate: function( nextProps ) {
+		return ! isEqual( nextProps.theme, this.props.theme );
```

Comment:
> Shouldn't comparing theme.id and theme.active be sufficient here?

Yeah you are right there, but the fact that there may not be a `theme` obj supplied (in the placeholder usage case) pushed me to go for `isEqual` as the simpler, more readable approach. Make sense, or should we just compare the relevant fields since this is on the speed-critical path?